### PR TITLE
Enable multi-tenant authentication with UAMI + auxiliary token provider

### PR DESCRIPTION
**Pull request by zarvd**
#### What type of PR is this?

/feature

#### What this PR does / why we need it:

This PR introduces a new authentication method using UAMI and an auxiliary token provider to support Managed Identity in multi-tenant scenarios, where the token is stored in Azure KeyVault.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Enable multi-tenant authentication with UAMI and auxiliary token provider
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/azclient/auth.go [hunk 1]
```diff
@@ -174,6 +174,8 @@ func NewAuthProvider(armConfig *ARMClientConfig, config *AzureAuthConfig, client
 		NetworkClientSecretCredential: networkClientSecretCredential,
 		NetworkTokenCredential:        networkTokenCredential,
 		MultiTenantCredential:         multiTenantCredential,
+
+		ClientOptions: clientOption,
```
Comment (zarvd) [inline pkg/azclient/auth.go:178]: NOTE: This change requires another PR to bump the azclient module.

File: pkg/azureclients/armauth/multi_tenant_token_provider.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+}
+
+func (p *MultiTenantTokenProvider) AuxiliaryOAuthTokens() []string {
+	p.logger.V(4).Info("Fetching auxiliary oauth token", "num-credentials", len(p.auxiliaryCredentials))
+	ctx, cancel := context.WithTimeout(context.Background(), p.timeout)
+	defer cancel()
+
+	var tokens []string
+	for _, cred := range p.auxiliaryCredentials {
+		token, err := cred.GetToken(ctx, policy.TokenRequestOptions{
```
Comment (nilo19) [inline pkg/azureclients/armauth/multi_tenant_token_provider.go:74]: Does it support list to reduce the cost?

Comment (zarvd) [reply pkg/azureclients/armauth/multi_tenant_token_provider.go:74]: No, the list would only contain 1 token provider at most in the long run; will optimize if the list grows.

**Comment by MartinForReal** [comment]
@zarvd Could you please rebase the branch? Thanks!

**Comment by zarvd** [comment]
rebased again

**Pull request merged**